core/src/main/scala/zio/config/ConfigSource.scala

```diff
@@ -53,16 +73,9 @@ object ConfigSource {
     map: Map[String, String],
     source: String = "constant",
     keyDelimiter: Option[Char] = None,
-    valueDelimter: Option[Char] = None
+    valueDelimiter: Option[Char] = None
   ): ConfigSource[String, String] =
-    fromMapInternal(map)(
-      x => {
-        val listOfValues = valueDelimter.fold(List(x))(delim => x.split(delim).toList)
-        ::(listOfValues.head, listOfValues.tail)
-      },
-      keyDelimiter,
-      source
-    )
+    fromMapInternal(map)(splitDelimited(_, valueDelimiter), keyDelimiter, source)
```

Comment:
I think, it makes sense to push this feature as a map for now. Let's go with incremental approach 👍 
However, we can revisit soon. I need to spend some time investing on PropertyTreeing it directly. I will let you know

Reply:
I am using `fromMapInternal` as a smart constructor of `ConfigSource` – the map only contains one key-value. The `ConfigSource`s are all then processed using `mergeAll[K, V](sources: Iterable[ConfigSource[K, V]])`.
Is there a better way of constructing a `ConfigSource`?